tests/app/notifications/test_validators.py

```diff
@@ -511,6 +440,96 @@ def test_should_not_rate_limit_if_limiting_is_disabled(
         assert not app.redis_store.exceeded_rate_limit.called
 
 
+def test_check_rate_limiting_checks_daily_limit_if_trial_mode_service(
```

Comment:
Minor: could we reduce the number of tests by just doing both assertions for the method calls in a single test, instead of spreading them across two? Also, we could combine them further by parametrizing on `restricted=True`. Then we'd just have one block of code, which would be much easier to read / review.

Reply:
refactor of the tests has been done

Reply:
Nice, thanks 🥇